Source/ProgressionSystemRuntime/Public/Components/PSHUDComponent.h

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "Data/PSTypes.h"
+#include "ProgressionSystemDataAsset.h"
+#include "Components/ActorComponent.h"
+#include "Structures/PlayerTag.h"
+//---
+#include "PSHUDComponent.generated.h"
+
+/**
+ * Implements the core logic on project about Progression System.
+ */
+
+UCLASS(Blueprintable, ClassGroup=(Custom), meta=(BlueprintSpawnableComponent))
+class PROGRESSIONSYSTEMRUNTIME_API UProgressionSystemComponent final : public UActorComponent
+{
+	GENERATED_BODY()
+
+public:
+	/** Sets default values for this component's properties. */
+	UProgressionSystemComponent();
+
+	/** Returns the Progression System data asset. */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	static const UProgressionSystemDataAsset* GetProgressionSystemDataAsset() { return &UProgressionSystemDataAsset::Get(); }
+
+	/** Returns current saved progression. */
+	UFUNCTION(BlueprintCallable, Category="C++")
+	FORCEINLINE FPSRowData GetSavedProgressionRowData() const { return SavedProgressionRowDataInternal; }
```

Comment:
Always return any structure by "const ref" (const FStruct&) as below:
`const FORCEINLINE FPSRowData& GetSavedProgressionRowData() const { return SavedProgressionRowDataInternal; }
So it does not copy the value anymore which is much cheaper

Reply:
done